main.tf

```diff
@@ -102,6 +102,30 @@ resource "aws_route_table_association" "private_routing_table" {
   count          = "${var.create_private_subnets ? length(var.availability_zones[var.aws_region]) : 0}"
 }
 
+data "aws_vpc_endpoint_service" "s3" {
+  count = "${var.create_s3_vpc_endpoint ? 1 : 0}"
+  service = "s3"
+}
+
+resource "aws_vpc_endpoint" "s3_vpc_endpoint" {
+  count = "${var.create_s3_vpc_endpoint ? 1 : 0}"
+  vpc_id = "${aws_vpc.vpc.id}"
+  service_name = "${data.aws_vpc_endpoint_service.s3.service_name}"
+  route_table_ids = ["${aws_route_table.public_routetable.id}", "${aws_route_table.private_routetable.id}"]
+  policy = <<POLICY
```

Comment:
@Casturan the policy is the default, also the policy option is optional. Did you give it a try without defining the policy?

Reply:
Without a policy it also works. I removed it.